python/benchmark/test_gen_data.py

```diff
@@ -258,7 +329,7 @@ def test_make_sparse_regression(
 
         # If there is no random shuffled redundant cols, we can check the total density
         density_num = float(density)
-        if redundant_cols == "0":
+        if redundant_cols == "0" and density_curve == "None":
```

Comment:
Seems only density_curve="None" is tested. How about density_curve="Linear" and "Exponential"?

Reply:
For linear and exponential, the thing is that after shuffling the matrix seems just purely random. Is there a good way to test it?